fs/unstable_copy_file_test.ts

```diff
@@ -0,0 +1,120 @@
+// Copyright 2018-2025 the Deno authors. MIT license.
+
+import { assert, assertRejects, assertThrows } from "@std/assert";
+import { mkdtemp, readFile, rm, stat, writeFile } from "node:fs/promises";
+import {
+  existsSync,
+  mkdtempSync,
+  readFileSync,
+  rmSync,
+  writeFileSync,
+} from "node:fs";
+import { tmpdir } from "node:os";
+import { join, resolve } from "node:path";
+import { copyFile, copyFileSync } from "./unstable_copy_file.ts";
+
+Deno.test("copyFile() copies content to an existed file", async () => {
+  const tempDirPath = await mkdtemp(resolve(tmpdir(), "copy_file_async_"));
+  const source = join(tempDirPath, "source.txt");
+  const target = join(tempDirPath, "target.txt");
+
+  const content = "This is written by `copyFile` API";
+
+  await writeFile(source, content);
+  await writeFile(target, "");
+  await copyFile(source, target);
+
+  const targetContent = await readFile(target, "utf-8");
+
+  assert(content === targetContent);
+
+  await rm(tempDirPath, { recursive: true });
```

Comment:
Let's use the following structure to avoid the leak of temp dirs:

```js
try {
  // ... test assertions
} catch {
  await rm(tempDirPath, { recursive: true });
}
```

The current test cases leave temp dirs unremoved when the test assertion failed for some reason.